amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/OptimizerGroupController.java

```diff
@@ -308,4 +315,8 @@ public void getContainers(Context ctx) {
                 .map(ContainerMetadata::getName)
                 .collect(Collectors.toList())));
   }
+
+  public static boolean validateGroupName(String groupName) {
```

Comment:
I thought of a more concise way, which is to put the group name pattern and existence in `validateGroupName` and throw an exception when appropriate.

```suggestion
  private void validateGroupName(String groupName) {
    // validate name pattern
    if (!GROUP_NAME_PATTERN.matcher(groupName).matches()) {
      throw new BadRequestException(...)
     }
   
      if (optimizerManager.getResourceGroup(name) != null) {
        throw new BadRequestException(String.format("Optimizer group:%s already existed.", name));
      }
  }
```